### PR TITLE
Add table splitting for PairPos format1

### PR DESCRIPTION
**Pull request by cmyr**
Built on top of #400, this adds the ability to split overflowing PairPos format 1 lookups.

I wanted to implement splitting of at least one lookup subtable on top of the promotion work in order to be confident that the approach I'd chosen for promotion would be suitable for splitting as well. Looks like it is?

### CODE REVIEW

File: write-fonts/src/graph/splitting.rs [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+}
+
+/// A common impl handling updating the lookuup with the new subtables
+fn split_subtables(
+    graph: &mut Graph,
+    lookup: ObjectId,
+    split_fn: fn(&mut Graph, ObjectId) -> Option<Vec<ObjectId>>,
+) {
+    let data = graph.objects.remove(&lookup).unwrap();
+    debug_assert!(data.reparse::<rgpos::PositionLookup>().is_ok());
```
Comment (dfrg) [inline write-fonts/src/graph/splitting.rs:28]: This function is called `split_subtables` and doesn’t seem to actually depend on it being a GPOS subtable. This feels like a bit of a footgun if/when we extend to GSUB.

File: write-fonts/src/graph/splitting.rs [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+fn split_pair_pos_subtable(graph: &mut Graph, lookup: ObjectId) -> Option<Vec<ObjectId>> {
+    let data = &graph.objects[&lookup];
+    let format: u16 = data.read_at(0).unwrap();
+    match format {
+        1 => split_pair_pos_format_1(graph, lookup),
+        2 => {
+            log::info!("table splitting not yet implemented for PairPos format 2");
+            None
+        }
+        _ => unreachable!("we only write valid formats"),
```
Comment (dfrg) [inline write-fonts/src/graph/splitting.rs:78]: Are we certain that this won’t be encountered when subsetting an arbitrary font? We don’t want to panic in that case. 

Maybe we can filter out subtables with invalid formats and emit a warning.

File: write-fonts/src/graph/splitting.rs [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+    if split_points.is_empty() {
+        return None;
+    }
+    split_points.push(data.offsets.len() - 1);
+    // okay, now we have a list of split points.
+
+    let mut new_subtables = Vec::new();
+    let mut prev_idx = 0;
+    for idx in &split_points {
```
Comment (dfrg) [inline write-fonts/src/graph/splitting.rs:135]: Small style nit: just consume the vec here? I don’t think it’s used after this.

**Review by dfrg — APPROVED** [review]
Seems a bit tricky in places so I’ve traced through the logic a few times and this looks solid to me.

A few comments inline but nothing major.

File: write-fonts/src/graph/splitting.rs (hunk 1 above)
Comment (cmyr) [reply write-fonts/src/graph/splitting.rs:27]: My understanding is that table splitting is only relevant for GPOS, where individual subtables can exceed 64k bytes. in GSUB the main problem is with having so many subtables that you get offset overflows, and that's handled by extension promotion.

That said I agree I should make that more explicit :)

**Pull request merged**